utils/wasm-builder/src/wasm_project.rs

```diff
@@ -64,6 +64,20 @@ pub struct WasmProject {
     profile: String,
     project_type: ProjectType,
     features: Option<Vec<String>>,
+    pre_processors: Vec<Box<dyn PreProcessor>>,
+}
+
+/// Preprocessor output
+pub struct PreProcessOutput {
+    /// Location of the artifact generated by pre-processor
+    pub path: PathBuf,
+    /// Content of the artifact generated by pre-processor
+    pub content: Vec<u8>,
+}
+
+/// Pre-processor hook for wasm generation
+pub trait PreProcessor {
+    fn pre_process(&self, original: PathBuf) -> Result<PreProcessOutput>;
```

Comment:
Why not `&Path` or `impl AsRef<Path>` instead of `Path`?

Reply:
because it can't be made into a trait object